scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -82,41 +182,38 @@ def write(self, outfile=None):
             with open(outfile,'w') as xmlout:
                 xmlout.write(xmlstr)
 
-    def get_node(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_node(self, nodename, attributes=None, root=None):
         """
         Get an xml element matching nodename with optional attributes.
 
         Error unless exactly one match.
         """
 
-        nodes = self.get_nodes(nodename, attributes=attributes, root=root, xpath=xpath)
+        nodes = self.get_nodes(nodename, attributes=attributes, root=root)
 
         expect(len(nodes) == 1, "Incorrect number of matches, {:d}, for nodename '{}' and attrs '{}' in file '{}'".format(len(nodes), nodename, attributes, self.filename))
         return nodes[0]
 
-    def get_optional_node(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_optional_node(self, nodename, attributes=None, root=None):
```

Comment:
I don't think we ever want users to use xpath as that leaks the fact that the underlying database is XML.